network/connection.go

```diff
@@ -614,29 +614,30 @@ func (conn *Connection) packetHandler() {
 		if pkt == nil {
 			return
 		}
-		// get handler
-		conn.Lock()
 
-		// execute handler or verdict
-		if conn.firewallHandler != nil {
-			conn.firewallHandler(conn, pkt)
-		} else {
-			defaultFirewallHandler(conn, pkt)
-		}
+		func() {
+			conn.Lock()
+			defer conn.Unlock()
 
```

Comment:
Performance wise I would not use an anonymous function here in the hot-path but rather use a normal, named function